Sources/MapboxDirections/Directions.swift

```diff
@@ -153,59 +116,86 @@ open class Directions: NSObject {
      - returns: The data task used to perform the HTTP request. If, while waiting for the completion handler to execute, you no longer want the resulting routes, cancel this task.
      */
     @discardableResult open func calculate(_ options: RouteOptions, completionHandler: @escaping RouteCompletionHandler) -> URLSessionDataTask {
-        let fetchStart = Date()
+        options.fetchStartDate = Date()
         let request = urlRequest(forCalculating: options)
         let requestTask = URLSession.shared.dataTask(with: request) { (possibleData, possibleResponse, possibleError) in
-            let responseEndDate = Date()
-            guard let response = possibleResponse, ["application/json", "text/html"].contains(response.mimeType) else {
+            
+            let offlineErrors: [URLError.Code] = [.cannotConnectToHost, .dataNotAllowed, .notConnectedToInternet]
```

Comment:
I don't think we should select only offline errors from all errors. IMO if `possibleError` is not nil, then we should call completion handler with some `.network(error)` and call it a day. 

Current solution still suffers from handling some URLSession errors as invalid response errors. E.g. if we cancel `URLSessionDataTask` returned by this method, then `possibleError` will be URLError with code `canceled`. Yet we'll call completion with `invalidResponse` error. This prevents users of this SDK from correctly handling request cancellation.

I propose putting `if let error = possibleError` at the beginning of the method, without any additional checks about the nature of the error, and just calling completion with error wrapped in our own `network` error case. That way we'll enable users of the SDK to correctly react to various URLSession errors.

Reply:
🤔 Agreed. Will change.